### PR TITLE
CHANGE (CodeAnalyzer): @W-13887486@: Make PMD7 the new default and remove PMD6

### PR DESCRIPTION
**Pull request by stephen-carter-at-sf**
* Remove PMD6 all together
* Use PMD 7.0.0 release now as default (updating inclusion list as well with new dependencies)
* Remove --preview-pmd7 flag
* Remove warnings about incompatible rules for PMD 7 (since we are now in PMD 7)
* Temporarily turn off unit test for pmd-appexchange (which will come in a change immediately following this one)
* Update smoke tests and sample code to reflect PMD 7

### CODE REVIEW

File: test/commands/scanner/run.filters.test.ts [hunk 1]
```diff
@@ -45,7 +45,9 @@ describe('scanner run tests that result in the use of RuleFilters', function ()
 				expect(csv.indexOf('\n')).to.equal(-1, "Should be no violations detected");
 			});
 
-			it('Case: --engine pmd-appexchange against unclean code', () => {
+			// Currently this test fails because the pmd-appexchange jar files depend on classes that only exist in PMD6
+			// TODO: Turn this test back on as soon as we get the new jar files for pmd-appexchange that work with PMD7
+			xit('Case: --engine pmd-appexchange against unclean code', () => {
```
Comment (stephen-carter-at-sf) [inline test/commands/scanner/run.filters.test.ts:50]: Will reenable this with PR: https://github.com/forcedotcom/sfdx-scanner/pull/1411 ... but this PR needs to be merged first.

File: package.json [hunk 2]
```diff
@@ -140,8 +140,8 @@
 		"prepack": "rm -rf lib && tsc -b && oclif manifest && oclif readme && oclif lock && npm shrinkwrap",
 		"postpack": "rm -f oclif.manifest.json oclif.lock npm-shrinkwrap.json",
 		"lint-typescript": "eslint ./src --ext .ts --max-warnings 0",
-		"test": "./gradlew test jacocoTestCoverageVerification && nyc mocha --timeout 10000 --retries 5 \"./test/**/*.test.ts\"",
-		"test-quiet": "cross-env SFGE_LOGGING=false ./gradlew test jacocoTestCoverageVerification && nyc mocha --timeout 10000 --retries 5 \"./test/**/*.test.ts\"",
+		"test": "./gradlew test jacocoTestCoverageVerification && nyc mocha --timeout 60000 --retries 5 \"./test/**/*.test.ts\"",
```
Comment (jag-j) [inline package.json:143]: why are we making this change?

File: pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/PmdRuleCataloger.java [hunk 3]
```diff
@@ -65,10 +65,10 @@ class PmdRuleCataloger {
 	void catalogRules() {
 		//Check for custom rules and process them
 
-		// STEP 1: Identify all of the ruleset and category files for each language we're looking at.
+		// Identify all the ruleset and category files for each language we're looking at.
 		extractRules();
 
-		// STEP 2: Process the category files to derive category and rule representations.
+		// STEP: Process the category files to derive category and rule representations.
```
Comment (jag-j) [inline pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/PmdRuleCataloger.java:71]: Nit: Remove "STEP:" so the comments are consistent.

File: package.json (hunk 2 above)
Comment (stephen-carter-at-sf) [reply package.json:143]: You'll notice that test-typescript already has a --timeout of 60000 ... so I was updating the others to reflect that change in all the commands. Basically on a given users machine a few tests go over 10 seconds... on my machine 3 tests consistently take 12 to 13 seconds. In the past, Josh updated test-typescript (which is the command he uses locally) to 60 seconds. I personally use yarn test locally to just run all the tests for all subprojects. So increasing this timeout to 60 seconds is the safest thing to do... it allows us to run things locally without tests dying but gives a sanity check for a run away test to stop and fail in our build system.

File: pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/PmdRuleCataloger.java (hunk 3 above)
Comment (stephen-carter-at-sf) [reply pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/PmdRuleCataloger.java:71]: ok - good catch.

File: pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/PmdRuleCataloger.java
[the remark this answers is not in the document]
Comment (stephen-carter-at-sf) [reply pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/PmdRuleCataloger.java:1]: ok - will do

File: sample-code/pmd-example-rules/pom.xml [hunk 4]
```diff
@@ -21,27 +21,10 @@ a JAR containing everything needed to register and run a custom rule.
...
 	<dependencies>
-		<dependency>
-			<groupId>net.sourceforge.pmd</groupId>
-			<artifactId>pmd</artifactId>
-			<version>6.55.0</version>
-			<type>pom</type>
-		</dependency>
 		<dependency>
 			<groupId>net.sourceforge.pmd</groupId>
 			<artifactId>pmd-apex</artifactId>
```
Comment (stephen-carter-at-sf) [inline sample-code/pmd-example-rules/pom.xml:26]: I just tested this locally and built it successfully. The only dependency needed is the pmd-apex v7.0.0 now.

**Review by jfeingold35 — APPROVED** [review]

**Review by jfeingold35 — APPROVED** [review]

**Comment by stephen-carter-at-sf** [comment]
Fixes #1293

**Pull request merged**